demos/linear_mixed_system/linear_mixed_system.py.rst

```diff
@@ -0,0 +1,328 @@
+
+    # eta
+    a_eta = trial_f * v_f * ds(top_id)
+    L_eta = eta * v_f * ds(top_id) + dt * dot( grad(v_f), grad(phi) ) * dx(fluid_id)
+    if coupling:
+        L_eta += - dt * dot( n_int, avg(U) ) * avg(v_f) * dS
+    LVP_eta = LinearVariationalProblem( a_eta, L_eta, result_mixed, bcs=BC_exclude_beyond_surface_mixed )
+    LVS_eta = LinearVariationalSolver( LVP_eta )
+
+Let us set the initial condition. We choose no motion at the beginning in both fluid and structure, zero displacement in the structure and deflected free surface in the fluid. The shape of the deflection is computed from the analytical solution::
+
+    # initial condition in fluid based on analytical solution
+    # compute analytical initial phi and eta
+    n_mode = 1
+    a = 0. * T / L**2 # in nondim units
+    b = 5. * T / L**2 # in nondim units
+    lambda_x = np.pi*n_mode/Lx
+    omega = np.sqrt( lambda_x*np.tanh(lambda_x*Lz) )
+    x = mesh.coordinates
+    phi_exact_expr = a * cos(lambda_x*x[0]) * cosh(lambda_x*x[1])
+    eta_exact_expr = - omega * b * cos(lambda_x*x[0]) * cosh(lambda_x*Lz)
+
+    bc_top = DirichletBC(V_W, 0, top_id)
+    eta.assign(0.)
+    phi.assign(0.)
+    eta_exact = Function(V_W)
+    eta_exact.interpolate( eta_exact_expr )
+    eta.assign( eta_exact, bc_top.node_set )
+    phi.interpolate( phi_exact_expr )
+    phi_f.assign( phi, bc_top.node_set)
+
+Files to store data for visualization::
+
+    outfile_phi = File("results_pvd/phi.pvd")
+    outfile_eta = File("results_pvd/eta.pvd")
+    outfile_X = File("results_pvd/X.pvd")
+    outfile_U = File("results_pvd/U.pvd")
+
+To save data for visualization, we change the position of the nodes in the mesh, so that they represent the computed dynamic position of the free surface and the structure::
+
+    def output_data():
+        output_data.counter += 1
+        if output_data.counter % output_data_every_x_time_steps != 0: return
+        mesh_static = mesh.coordinates.vector().get_local()
+        mesh.coordinates.vector().set_local( mesh_static + X.vector().get_local() )
+        mesh.coordinates.dat.data[:,1] += eta.dat.data_ro
+        outfile_phi.write( phi )
+        outfile_eta.write( eta )
+        outfile_X.write( X )
+        outfile_U.write( U )
+        mesh.coordinates.vector().set_local( mesh_static )
+    output_data.counter = -1 # -1 to exclude counting print of initial state
+
+In the end, we proceed with the actual computation loop::
+
+    t = 0.
+    output_data()
+
+    while t <= t_end + dt:
+        t += dt
+        print 'time = ', t * T
+        # symplectic Euler scheme
+        LVS_phi_f.solve()
+        LVS_U_phi.solve()
+        tmp_f, tmp_s = result_mixed.split()
+        phi.assign(tmp_f)
+        U.assign(tmp_s)
+        LVS_eta.solve()
+        tmp_f, _ = result_mixed.split()
+        eta.assign(tmp_f)
+        LVS_X.solve()
+
+        output_data()
+
+The result of the computation, visualised with `paraview <http://www.paraview.org/>`__, is shown below.
```

Comment:
What would be really cool at this point would be a youtube video of the solution. You can see how to embed videos by looking in `docs/source/index.rst`.